packages/starknet-snap/src/state/transaction-state-manager.ts

```diff
@@ -0,0 +1,305 @@
+  contractAddress?: string[];
+  executionStatus?: TransactionExecutionStatus[];
+  finalityStatus?: TransactionFinalityStatus[];
+  timestamp?: number;
+};
+
+export class TransactionStateManager extends StateManager<Transaction> {
+  protected getCollection(state: SnapState): Transaction[] {
+    return state.transactions;
+  }
+
+  protected updateEntity(dataInState: Transaction, data: Transaction): void {
+    assert(
+      data.executionStatus,
+      enums(Object.values(TransactionExecutionStatus)),
+    );
+    assert(
+      data.finalityStatus,
+      enums(Object.values(TransactionFinalityStatus)),
+    );
+    assert(data.timestamp, number());
+
+    dataInState.executionStatus = data.executionStatus;
+    dataInState.finalityStatus = data.finalityStatus;
+    dataInState.timestamp = data.timestamp;
+    dataInState.failureReason = data.failureReason;
+  }
+
+  #getCompositeKey(data: Transaction): string {
+    const key1 = BigInt(data.chainId);
+    const key2 = BigInt(data.txnHash);
+    return `${key1}&${key2}`;
+  }
+
+  async findTransactions(
+    {
+      txnHash,
+      txnType,
+      chainId,
+      senderAddress,
+      contractAddress,
+      executionStatus,
+      finalityStatus,
+      timestamp,
+    }: SearchFilter,
+    state?: SnapState,
+  ): Promise<Transaction[]> {
+    const filters: ITxFilter[] = [];
+    if (txnHash !== undefined && txnHash.length > 0) {
+      filters.push(new TxHashFilter(txnHash));
+    }
+
+    if (chainId !== undefined && chainId.length > 0) {
+      filters.push(new ChainIdFilter(chainId));
+    }
+
+    if (timestamp !== undefined) {
+      filters.push(new TxTimestampFilter(timestamp));
+    }
+
+    if (senderAddress !== undefined && senderAddress.length > 0) {
+      filters.push(new SenderAddressFilter(senderAddress));
+    }
+
+    if (contractAddress !== undefined && contractAddress.length > 0) {
+      filters.push(new ContractAddressFilter(contractAddress));
+    }
+
+    if (txnType !== undefined && txnType.length > 0) {
+      filters.push(new TxnTypeFilter(txnType));
+    }
+
+    if (finalityStatus !== undefined || executionStatus !== undefined) {
+      filters.push(
+        new TxStatusFilter(finalityStatus ?? [], executionStatus ?? []),
+      );
+    }
+
+    return this.list(
+      filters,
+      // sort by timestamp in descending order
+      (entityA: Transaction, entityB: Transaction) =>
+        entityB.timestamp - entityA.timestamp,
+      state,
+    );
+  }
+
+  /**
+   * Finds a transaction object based on the given chain id and txn hash.
+   *
+   * @param param - The param object.
+   * @param param.txnHash - The txn hash of the transaction object to search for.
+   * @param [param.chainId] - The optional chain id of the transaction object to search for.
+   * @param [state] - The optional SnapState object.
+   * @returns A Promise that resolves with the transaction object if found, or null if not found.
+   */
+  async findTransaction(
```

Comment:
maybe we call this getTransaction, as it is very different from findTransactions above. Having similar name could be confusing.

Reply:
yup, so lets change all in other state manager too